iree/compiler/Dialect/Flow/Transforms/DispatchLinalgOnTensors.cpp

```diff
@@ -849,6 +849,23 @@ static void decideFusableLinalgOps(FuncOp funcOp) {
                             builder.getI64ArrayAttr(fusionGroups));
       }
     }
+
+    // As a second step mark all the element-wise linalg ops not fused as roots
+    // so that they get tiled and distributed.
+    for (linalg::LinalgOp linalgOp : linalgOps) {
```

Comment:
So the only reason we have the `MakeDispatchWorkgroupsOp` pattern is for the fall back right? In other words in the `ConvertToGPU` pass, the mapping of iterations of `linalg` operation to global invocation ID is not used in the Linalg to tensors path. That would be great, cause then we can move all that loop distribution logic all the way in Flow.

Reply:
Correct, there is still few cases that don't tile, I can look at those at some point. Yes for all those cases the `ConverToGPU` global invocation ID is not reached. It would be good to do the loop distribution in flow indeed. 
The main case I have seen remaining is when the linalg affine map are not projected permutation.